superset-frontend/src/datasource/DatasourceEditor.jsx

```diff
@@ -485,11 +486,19 @@ class DatasourceEditor extends React.PureComponent {
 
   syncMetadata() {
     const { datasource } = this.state;
-    const endpoint = `/datasource/external_metadata_by_name/${
-      datasource.type || datasource.datasource_type
-    }/${datasource.database.database_name || datasource.database.name}/${
-      datasource.schema
-    }/${datasource.table_name}/`;
+    const params = {
+      datasource_type: datasource.type || datasource.datasource_type,
+      database_name:
+        datasource.database.database_name || datasource.database.name,
+      schema_name: datasource.schema,
+      table_name: datasource.table_name,
+    };
+    const endpoint = `/datasource/external_metadata_by_name/?q=${rison.encode(
+      // rison can't encode the undefined value
+      Object.keys(params).map(key =>
+        params[key] === undefined ? null : params[key],
+      ),
```

Comment:
Don't we need to encode the key/value pairs (`Object.entries`), not just the values (`Object.keys`)? (honest question, I haven't done a lot of this rison API stuff)

Reply:
this is to code entire `params` field, just change `undefined` to `null`, because prison can't encode undefined value

Reply:
Ah yes now I see! 🤦